awpy/analytics/map_control.py

```diff
@@ -0,0 +1,409 @@
+from collections import defaultdict
+import numpy as np
+
+from awpy.data import NAV, NAV_GRAPHS
+from awpy.analytics.nav import area_distance, find_closest_area, calculate_tile_area
+from awpy.types import GameRound, GameFrame, TeamFrameInfo, FrameMapControl
+
+
+def approx_neighbors(
+    map_name: str,
+    source_tile_id: int,
+    n: int = 5,
+) -> list[tuple[float, int]]:
+    """Approximates neighbors for tiles by finding n closest. Helpful for
+       tiles who are 'isolated' (they have no navigable tiles as a neighbor).
+    Args:
+        map_name (string)   : Map for source_tile_id
+        source_tile_id (int): Area ID as an integer
+        n (int)             : Number of closest tiles/approximated
+                              neighbors wanted
+    Returns:
+        List of tuples representing n closest tiles.
+            Tuple[0] -> Distance between source_tile_id and represented tile
+            Tuple[1] -> Area ID of represented tile
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+                    If source_tile_id is not in awpy.data.NAV[map_name]
+                    If the length of point is not 3
+    """
+
+    if map_name not in NAV:
+        raise ValueError("Map not found.")
+    if source_tile_id not in NAV[map_name].keys():
+        raise ValueError("Area ID not found.")
+    if n < 0:
+        raise ValueError("n must be >= 0")
+
+    currentMapInfo = NAV[map_name]
+    distanceArr = []
+
+    for tile in currentMapInfo:
+        if tile != source_tile_id:
+            currentDistance = area_distance(
+                map_name, tile, source_tile_id, dist_type="euclidean"
+            )["distance"]
+            distanceArr.append((currentDistance, tile))
+    distanceArr.sort()
+
+    return distanceArr[:n]
+
+
+def bfsHelper(
```

Comment:
Linters will complain about function and variable names that are not `snake_case`

Reply:
Addressed in next commit on branch